dietpi/dietpi-software

```diff
@@ -533,22 +540,135 @@ INDEX_WEBSERVER_TARGET=${INDEX_WEBSERVER_TARGET:=-2}
 		#--------------------------------------------------------------
 		#Install Software
 
-		#Installs that require DESKTOP_LXDE
+		#Installs that require a DESKTOP
 		if (( $TIGHTVNCSERVER == 1 )); then
-			DESKTOP_LXDE=1
+			if (( $DESKTOP_LXDE == 0 )) && (( $DESKTOP_MATE == 0 )) && (( $DESKTOP_GNUSTEP == 0 )); then
```

Comment:
I agree about implementing the Desktop Preference system at some point, @Fourdee.

And @rhkean -- still no coffee yet :-)  I will discuss more on #41 how much I agree with the idea of implementing a Desktop Preference system.

To remain on point, I took your pointers and re-tested my code.  This came to a work around as my nest of if/then conditions was a bloody mess.  I simplified it and re-tested by selecting all sorts of packages that would depend on a Desktop Environment to be chosen.

During the install process, naturally not DESKTOP_LXDE, MATE, nor GNUStep were selected.  When it came time to look for a DESKTOP to install, I added a prompt -- since LXDE is the original light-weight Desktop using whiptail:

![desktop_warning](https://cloud.githubusercontent.com/assets/10519408/15636506/dbd42634-25ce-11e6-8c7e-b4c577d11622.PNG)

After pressing enter -- presto!  The installer continues with LXDE and all remaining GUI-Centric-Software (QuiteRSS, etc) without fault or being confused.

On reboot, I validated the results:

![post-workaround](https://cloud.githubusercontent.com/assets/10519408/15636510/01278f5c-25cf-11e6-8f31-5c4e9e4975d3.PNG)

So, this isn't permanent -- but I look forward to feedback on this for now as I continue along this commit/pull.